toolchain/semantics/semantics_context.cpp

```diff
@@ -604,13 +628,42 @@ auto SemanticsContext::CanonicalizeStructType(ParseTree::Node parse_node,
       parse_node, SemanticsTypeId::TypeType, refs_id));
   auto type_id = semantics_ir_->AddType(node_id);
   CARBON_CHECK(canonical_types_.insert({node_id, type_id}).second);
-  canonical_struct_types_nodes_.push_back(
-      std::make_unique<StructTypeNode>(canonical_id, type_id));
-  canonical_struct_types_.InsertNode(canonical_struct_types_nodes_.back().get(),
+  canonical_types_nodes_.push_back(
+      std::make_unique<TypeNode>(canonical_id, type_id));
+  canonical_struct_types_.InsertNode(canonical_types_nodes_.back().get(),
                                      insert_pos);
   return type_id;
 }
 
+auto SemanticsContext::CanonicalizeTupleType(
+    ParseTree::Node parse_node,
+    const llvm::SmallVector<SemanticsTypeId>&& type_ids) -> SemanticsTypeId {
```

Comment:
This should be a non-const rvalue reference so that we can move from it.
```suggestion
    llvm::SmallVector<SemanticsTypeId>&& type_ids) -> SemanticsTypeId {
```